src/cache-restore.ts

```diff
@@ -29,7 +29,9 @@ export const restoreCache = async (
     );
   }
 
-  const primaryKey = `setup-go-${platform}-go-${versionSpec}-${fileHash}`;
+  const linuxVersion =
+    (process.env.RUNNER_OS === 'Linux' && `${process.env.ImageOS}-`) || '-';
```

Comment:
You need to change `${process.env.ImageOS}-` because it won't be falsy value

Reply:
We can keep it if we're sure that ImageOS persist with RUNNER_OS

Reply:
I want to check for "${process.env.ImageOS}-" is not false for sake of self-hosted environments. The probability of RUNNER_OS is set and ImageOS is not set is low but still above zero.